throughput/src/bin/z_sub_thr.rs

```diff
@@ -88,9 +93,12 @@ async fn main() {
     let c_messages = messages.clone();
 
     let session = zenoh::open(config).await.unwrap();
-    let _subscriber = session
-        .subscribe(KEY_EXPR)
-        .callback(move |_| {
+    let _subscriber = if use_expr {
```

Comment:
Maybe preventing the ambiguous type after an if/else branch is better. For example, the callback function is associated with the struct SubscriberBuilder, so we may write down them clearly instead of a plain if/else branch. 

There are two advantages of writing in this way:

```rust
    let session = zenoh::open(config).await.unwrap();
    let _subscriber = {
        let builder = if use_expr {
            session.subscribe(KEY_EXPR)
        } else {
            session.subscribe(session.declare_expr(KEY_EXPR).await.unwrap())
        };
        builder
            .callback(move |_| {
                c_messages.fetch_add(1, Ordering::Relaxed);
            })
            .reliable()
            .push_mode()
            .await
            .unwrap()
    };
```

1.  Explicit info with Rust analyzer
![Untitled](https://user-images.githubusercontent.com/10692887/157831024-888e1db5-11f1-4552-ad09-71a008cd4cf4.png)

2.  Better folding
![Untitled](https://user-images.githubusercontent.com/10692887/157831113-aa21bf22-ea23-4f38-8454-d28337614e32.png)

BTW, we need to specify reliable and push mode to be consistent with zn_sub_thr.rs

Reply:
Yes. At the beginning I separated the builder into two parts, but later decided to combine them together. It did look better and can help increase the readability, especially for the case that the builder types are different. And, thanks for pointing out the missed reliable and push mode settings. Will re-submit it later.